### ISSUE TITLE
Import via bluetooth fails on Mares Puck Pro + 

### ISSUE THREAD

**Issue by janis1995** [Subsurface-divelog/subsurface#4386]
### Describe the issue:

- [x] Bug
- [ ] Change request
- [ ] New feature request
- [ ] Discussion request

### Issue long description:

I'm trying to import my dive log from a Mares Puck Pro + Dive Computer via bluetooth. The dive log contains 49 dives.
After importing 26 dives the import aborts and the dive computer ends up in its boot mode.
Reflashing firmware or removing battery is required for recovery of the dive computer.
If I try to connect and download after recovery again. Subsurface says no more new dives available on the computer.

### Operating system:

OS: Android 15 

Pixel 6A

### Subsurface version:

6.0.5276 (Official Release)
Installation via APK as the App can't be found in the Google Play Store for Android 15 devices.

### Steps to reproduce:

1) Pair the phone with the dive computer
2) Open the Subsurface App
3) Connect to the dive computer
4) Start downloading dives 

### Current behavior:

Download aborts always after importing 26 dives and dive computer ends up in boot mode.
Issue was reproduced multiple times. 

### Expected behavior:

Download finishes for all dives. It would be also nice to be able to select which dives to be imported.

### Additional information:

Mares Puck Pro + Firmware:  08.00.01

I would have attached the Log. But I can't find any app in android which is able to handle that much data from the clipboard.

**Comment by jefdriesen** [comment]
Subsurface mobile can generate an email with the log files as attachment, which you can send to yourself. The steps are explained here:
https://libdivecomputer.org/subsurface.html#mobile

**Comment by janis1995** [comment]
Thanks for the hint this time even more dives got imported, but still the dive computer ends in boot mode.
If I can assist in debugging, please let me know I have some experience in embedded C development ;)
[libdivecomputer.log](https://github.com/user-attachments/files/18039529/libdivecomputer.log)
[subsurface.log](https://github.com/user-attachments/files/18039530/subsurface.log)

**Comment by jefdriesen** [comment]
Unfortunately the subsurface log is missing all the important info. You'll need to try with a build that includes this fix: https://github.com/subsurface/subsurface/pull/4385

**Comment by janis1995** [comment]
I have retested it with version 6.0.5298!
[libdivecomputer.log](https://github.com/user-attachments/files/18135315/libdivecomputer.log)
[subsurface.log](https://github.com/user-attachments/files/18135316/subsurface.log)

**Comment by jefdriesen** [comment]
Sorry, I completely lost track of this ticket.

In both logs, the download goes fine until it suddenly fails (respectively around 72% and 49%). At that point the dive computer doesn't answer anymore. Unfortunately subsurface disables the detailed debug logging after the first few BLE data packets:

https://github.com/subsurface/subsurface/blob/master/core/qt-ble.cpp#L740-L743

So I don't know what went wrong. Do you still see the problem with the latest version?

I have a custom subsurface build with this logging limitation removed. Can you maybe try that? You can download it here:

https://github.com/jefdriesen/subsurface/actions/runs/14912639529
https://libdivecomputer.org/builds/subsurface/

(To download build artifacts from github you need an account. If you don't have one, you can use the second link.)

**Comment by janis1995** [comment]
Issue is still happening. Sharing the logs made with the special subsurface version.
Always when the issue happens the dive computer ends up in bootloader mode, so could be a problem of the dive computer firmware itself aswell.

[subsurface (1).log](https://github.com/user-attachments/files/20838174/subsurface.1.log)
[libdivecomputer (1).log](https://github.com/user-attachments/files/20838173/libdivecomputer.1.log)

**Comment by jefdriesen** [comment]
Unfortunately the log with the custom build doesn't show anything new. The download works just fine, with no errors whatsoever, until we hit that first timeout. Retrying didn't help, so it's immediately a fatal error. In most other Mares logs, I typically see a large number of errors where retrying usually works before it finally hits a fatal one. For those cases, I just submitted PR #4543. But in your log, I don't seen anything where those changes would help.

Does the dive computer screen show anything when it jumps to bootloader? I guess that means the firmware crashed. Can you maybe record a video of the dive computer?

**Comment by janis1995** [comment]
Yes I assume firmware is crashing. Even I wrote to Mares, there reply was like just don't use subsurface for logging dives :/ 
So seems a bug in the puck pro+ I would say. Will close the issue!
